src/test/java/org/logstash/input/DeadLetterQueueInputPluginTests.java

```diff
@@ -112,4 +112,28 @@ public void testTimestamp() throws Exception {
         DeadLetterQueueInputPlugin plugin = new DeadLetterQueueInputPlugin(dir, false, null, new Timestamp(targetDateString));
         plugin.register();
     }
+
+    @Test
+    public void testClosingEmptyDlq() throws Exception {
+        Path since = temporaryFolder.newFile(".sincdb").toPath();
+        DeadLetterQueueInputPlugin plugin = new DeadLetterQueueInputPlugin(dir, true, since, null);
+
+        final AtomicInteger count = new AtomicInteger();
+        Thread pluginThread = new Thread(() -> {
+            try {
+                plugin.register();
+                plugin.run((e) -> {count.incrementAndGet();});
+            } catch (Exception e) {
+                // do nothing
+            }
+        });
+        pluginThread.start();
+        Thread.sleep(50);
```

Comment:
Which is the motivation for the `Thread.sleep(50)` ?

Reply:
As you may have seen I took some inspiration from one of the other tests in writing this, and I was uncertain if there was thread-related reasons for sleeping a little bit before interrupting it. So I left it in originally and then I never came back to it. However, I just tested and it works without the sleep too, so I've removed it in the last push.